Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -219,6 +219,32 @@ public struct AbstractInterpreter {
         setType(of: v, to: t)
     }
 
+    private func calleeTypes(for signature: FunctionSignature) -> [Type] {
+
+        func processType(_ type: Type) -> Type {
+            if type == .anything {
+                return .unknown
+            } else if type == .iterable {
+                return environment.arrayType
+            } else {
+                return type
+            }
+        }
+
+        var types: [Type] = []
+        signature.parameters.forEach { param in
+            switch param {
+                case .plain(let t):
+                    types.append(processType(t))
+                case .opt(let t):
+                    types.append(processType(t) | .undefined)
```

Comment:
I think this should be `processType(t | .undefined)`, so that `.opt(.anything)` just turns into `.unknown` and not `.unknown | .undefined` (`.unknown` already means that we don't know what it is, so adding in `.undefined` doesn't really make sense and might screw other code that checks for `.unknown`). Maybe make a comment about this behaviour, since it's another tricky bit of logic necessary due to the distinction between `.anything` and `.unknown`...

Reply:
I'll create an issue about the `.unknown` vs. `.anything` weirdness